demos/http/http_demo_plaintext/http_demo_plaintext.c

```diff
@@ -0,0 +1,552 @@
+
+    if( httpStatus == HTTP_SUCCESS )
+    {
+        /* Initialize the response object. The same buffer used for storing
+         * request headers is reused here. */
+        response.pBuffer = userBuffer;
+        response.bufferLen = USER_BUFFER_LENGTH;
+
+        LogInfo( ( "Sending HTTP %s request to %s%s...",
+                   pMethod, SERVER_HOST, pPath ) );
+        LogInfo( ( "Request Headers:\n%.*s",
+                   ( int32_t ) requestHeaders.headersLen,
+                   ( char * ) requestHeaders.pBuffer ) );
+        LogInfo( ( "Request Body:\n%.*s\n",
+                   ( int32_t ) REQUEST_BODY_TEXT_LENGTH,
+                   ( char * ) requestBodyBuffer ) );
+        /* Send the request and receive the response. */
+        httpStatus = HTTPClient_Send( pTransportInterface,
+                                      &requestHeaders,
+                                      requestBodyBuffer,
+                                      REQUEST_BODY_TEXT_LENGTH,
+                                      &response,
+                                      0 );
+    }
+
+    if( httpStatus == HTTP_SUCCESS )
+    {
+        LogInfo( ( "Received HTTP response from %s%s...",
+                   SERVER_HOST, pPath ) );
+        LogInfo( ( "Response Headers:\n%.*s",
+                   ( int32_t ) response.headersLen,
+                   response.pHeaders ) );
+        LogInfo( ( "Response Status:\n%u",
+                   response.statusCode ) );
+        LogInfo( ( "Response Body:\n%.*s\n",
+                   ( int32_t ) response.bodyLen,
+                   response.pBody ) );
+    }
+    else
+    {
+        LogError( ( "Sending HTTP %s request to %s%s failed with status = %u.",
+                    pMethod, SERVER_HOST, pPath, httpStatus ) );
+    }
+
+    if( httpStatus == HTTP_SUCCESS )
+    {
+        return EXIT_SUCCESS;
+    }
+    else
+    {
+        return EXIT_FAILURE;
+    }
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Entry point of demo.
+ */
+int main()
+{
+    int returnStatus = EXIT_SUCCESS;
+    HTTPNetworkContext_t socketContext = { 0 };
+    HTTPTransportInterface_t transportInterface = { 0 };
+
+    /* Set the request body. */
+    strncpy( ( char * ) requestBodyBuffer,
+             REQUEST_BODY_TEXT, REQUEST_BODY_TEXT_LENGTH );
+
+    /**************************** Connect. ******************************/
+
+    /* Establish TCP connection. */
+    returnStatus = connectToServer( SERVER_HOST, SERVER_PORT, &socketContext.tcpSocket );
+
+    /* Define the transport interface. */
+    if( returnStatus == EXIT_SUCCESS )
+    {
+        transportInterface.recv = transportRecv;
+        transportInterface.send = transportSend;
+        transportInterface.pContext = &socketContext;
+    }
+
+    /*********************** Send HTTPS request. ************************/
+
+    /* The client is now connected to the server. This example will send a
```

Comment:
We  need to have detailed comments explaining what each sendHttpRequest() is doing and what is end result. In general please look how comments are written in MQTT demos for the flow of the demo:
https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/development/demos/mqtt/mqtt_demo_plaintext/mqtt_demo_plaintext.c#L816